rust/src/operations/transaction/state.rs

```diff
@@ -82,6 +84,45 @@ impl DeltaTableState {
 
         Ok(sql_to_rel.sql_to_expr(sql, &df_schema, &mut Default::default())?)
     }
+
+    /// Get the pysical table schema.
+    ///
+    /// This will construct a schema derived from the parqet schema of the latest data file,
+    /// and fields for partition columns from the schema defined in table meta data.
+    pub async fn physical_arrow_schema(
```

Comment:
I don't think we can guarantee this schema is consistent across all parquet files in the table; different writers may have written to the table with different physical types for timestamps. IMO this should be handled in the scan of each Parquet file. That is, we should cast the physical type to microsecond timestamps as needed.

Reply:
In PyArrow, we handle the int96 timestamp issue by passing an argument to the reader to coerce it to microsecond precision. Maybe we could implement something similar upstream?

https://github.com/delta-io/delta-rs/blob/34d43b6b336c065731dca4c43268ac18b0beef9c/python/tests/test_table_read.py#L32

Reply:
There definitely are no guarantees as to the file schema being consistent. Datafusion however needs a consistent schema. Once we get into column mappings etc, things might get even more demanding and we may have to roll our own parquet scan, or rather start putting logic into our DeltaScan.

That said, I do believe using the schema from the latest file is an improvement over the current way, which at least for me fails for more or less every databricks written table where there are timestamps involved.

Not sure about the best way forward, but I'm happy to keep that logic on a private branch somewhere until we have a more general fix.

Reply:
somewhat related, so I already published it as WIP - in #1267 I did some work on the write command. There my plan was to use the same schema to validates writes. But there it would be even more confusing, since we might end up on situation, where writing the "official" schema of the chart would not be permissible. But somehow it feels very strange to me to have potentially many schemas in the same table.

i guess spark must allow at least some flexibility in what schema it expects at write time, otherwise how would we end up in this discussion at all :D.

Reply:
Yeah we are definitely hitting the limits of DataFusion's scanner. I've created an issue upstream https://github.com/apache/arrow-datafusion/issues/5950

I'm fine with moving this forward; I mostly care that we have a more robust implementation in the future and have at least some momentum towards it.